Muiv5 upgrade

aachavarria opened the pull request:
Confirm Dropdown
RenderTreeNodes
PathNavigatorTree: Fixing issues related to onLabel and onIcon click;
Fixing issues related to background color paper on inputs and dialog;

aachavarria committed:
Merge branch 'muiv5-upgrade' of https://github.com/rart/studio-ui into muiv5-upgrade

aachavarria committed:
Merge branch 'muiv5-upgrade' of https://github.com/rart/studio-ui into muiv5-upgrade

aachavarria committed:
Fixing styles on PathNavigatorTree when using material ui 5;

aachavarria committed:
Merge branch 'muiv5-upgrade' of https://github.com/rart/studio-ui into muiv5-upgrade

aachavarria committed:
fixing background.default color;
fixing issues on RenderTreeNode.tsx related to onIconClick and onLabelClick

aachavarria committed:
adding background paper color

aachavarria committed:
creating iconButton component

aachavarria committed:
adding color="inherit" to secondaryButton;
fixing button color on ItemStatesManagement.tsx;

aachavarria committed:
pending discussion about inputFormat

aachavarria committed:
adding EnhancedDialog to AuditLogEntryParametersDialog.tsx

aachavarria committed:
adding EnhancedDialog to BrowseFilesDialog.tsx

aachavarria committed:
using asLocalizedDateTime on DateTimePicker.tsx

aachavarria committed:
adding backgroundColor paper to MuiInputBase and MuiOutlinedInput

aachavarria committed:
adding missing effect dependency browseFilesDialogState
removing handleClick function on ConfirmDropdown.tsx

rart commented on ui/app/src/components/Controls/ConfirmDropdown.tsx:
```diff
@@ -104,19 +98,28 @@ export default function ConfirmDropdown(props: ConfirmDropdownProps) {
     onCancel?.();
   };
 
+  const iconButton = useMemo(
```
```suggestion
    // Not memoizing causes the menu to get misplaced upon opening.
    const iconButton = useMemo(
```

rart commented on ui/app/src/components/Dialogs/DialogBody.tsx:
```diff
@@ -29,7 +29,7 @@ const styles = makeStyles((theme: Theme) =>
       flex: '1 1 auto',
       flexDirection: 'column',
-      padding: theme.spacing(2),
+      padding: `${theme.spacing(2)} !important`,
```
This needs to change to the emotion-based API

aachavarria replied:
I will work this after merge, with another pending changes

rart commented on ui/app/src/components/FolderBrowserTreeView/RenderTreeNode.tsx:
```diff
@@ -44,16 +47,15 @@ export function RenderTreeNode(props: RenderTreeNodeProps) {
-      // TODO: Handle label/icon click
-      // onIconClick={(e) => onIconClick?.(e, node)}
-      // onLabelClick={(e) => onLabelClick?.(e, node)}
+      expandIcon={<ExpandMoreIcon onClick={(e) => onIconClick?.(e, node)} />}
+      collapseIcon={<ChevronRightIcon onClick={(e) => onIconClick?.(e, node)} />}
```
add role="button" to icons

rart commented on ui/app/src/components/FolderBrowserTreeView/RenderTreeNode.tsx:
```diff
@@ -44,16 +47,15 @@ export function RenderTreeNode(props: RenderTreeNodeProps) {
       key={node.id}
       nodeId={node.id}
-      label={node.name}
+      label={<div onClick={(e) => onLabelClick?.(e, node)}>{node.name}</div>}
```
add role=button

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTreeItem.tsx:
```diff
@@ -84,7 +90,8 @@ const useStyles = makeStyles((theme) =>
       display: 'flex',
       alignItems: 'center',
-      height: '26px',
+      // height: '26px',
```
x

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTreeUI.tsx:
```diff
@@ -130,13 +128,7 @@ export default function PathNavigatorTreeUI(props: PathNavigatorTreeUIProps) {
-          defaultCollapseIcon={<ArrowDropDownRoundedIcon />}
-          disableSelection={true}
-        >
+        <TreeView className={classes.root} expanded={expandedNodes} disableSelection={true}>
```
```suggestion
        <TreeView className={classes.root} expanded={expandedNodes} disableSelection>
```

rart commented on ui/app/src/components/SiteConfigurationManagement/SiteConfigurationManagement.tsx:
```diff
@@ -585,7 +585,7 @@ export default function SiteConfigurationManagement(props: SiteConfigurationMana
                       {formatMessage(translations.encryptMarked)}
                     </SecondaryButton>
-                    <Button size="small" onClick={onEncryptHelpClick}>
+                    <Button size="small" color="inherit" onClick={onEncryptHelpClick}>
```
What's the inherit for?

aachavarria committed:
adding comment to iconButton memo;
adding role="button: to icons on RenderTreeNode;
removing inherit from button on SiteConfigurationManagement.tsx;

aachavarria committed:
adding EnhancedDialog to DeleteContentTypeDialog

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTreeItem.tsx:
```diff
@@ -245,17 +257,21 @@ export default function PathNavigatorTreeItem(props: PathNavigatorTreeItemProps)
           label={
             <>
-              <section className={classes.itemDisplaySection} onMouseOver={onMouseOver} onMouseLeave={onMouseLeave}>
+              <section
+                onClick={(event) => onLabelClick(event, node.id)}
+                className={classes.itemDisplaySection}
+                onMouseOver={onMouseOver}
+                onMouseLeave={onMouseLeave}
+              >
```
```suggestion
              <section
                role="button"
                onClick={(event) => onLabelClick(event, node.id)}
                className={classes.itemDisplaySection}
                onMouseOver={onMouseOver}
                onMouseLeave={onMouseLeave}
              >
```

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTreeItem.tsx:
```diff
@@ -245,17 +257,21 @@ export default function PathNavigatorTreeItem(props: PathNavigatorTreeItemProps)
-          // TODO: replace onLabelClick & onIconClick
-          // onLabelClick={(event) => onLabelClick(event, node.id)}
-          // onIconClick={() => onIconClick(node.id)}
+          expandIcon={<ArrowRightRoundedIcon onClick={() => onIconClick(node.id)} />}
+          collapseIcon={<ArrowDropDownRoundedIcon onClick={() => onIconClick(node.id)} />}
```
```suggestion
          expandIcon={<ArrowRightRoundedIcon role="button" aria-label={formatMessage(translations.expand)} aria-hidden="false" onClick={() => onIconClick(node.id)} />}
          collapseIcon={<ArrowDropDownRoundedIcon role="button" aria-label={formatMessage(translations.collapse)} aria-hidden="false" onClick={() => onIconClick(node.id)} />}
```

rart commented on ui/app/src/components/FolderBrowserTreeView/RenderTreeNode.tsx:
```diff
@@ -44,16 +47,19 @@ export function RenderTreeNode(props: RenderTreeNodeProps) {
-      // TODO: Handle label/icon click
-      // onIconClick={(e) => onIconClick?.(e, node)}
-      // onLabelClick={(e) => onLabelClick?.(e, node)}
+      expandIcon={<ExpandMoreIcon role="button" onClick={(e) => onIconClick?.(e, node)} />}
+      collapseIcon={<ChevronRightIcon role="button" onClick={(e) => onIconClick?.(e, node)} />}
```
```suggestion
      expandIcon={<ExpandMoreIcon role="button" aria-label={formatMessage(translations.expand)} aria-hidden="false" onClick={(e) => onIconClick?.(e, node)} />}
      collapseIcon={<ChevronRightIcon role="button" aria-label={formatMessage(translations.collapse)} aria-hidden="false" onClick={(e) => onIconClick?.(e, node)} />}
```

rart commented on ui/app/src/components/DeleteContentTypeDialog/DeleteContentTypeDialogContainer.tsx:
```diff
@@ -0,0 +1,80 @@
+    onSubmittingAndOrPendingChange({
+      isSubmitting: true
+    });
+    deleteContentType(site, contentType.id).subscribe(
+      () => {
+        onSubmittingAndOrPendingChange({
+          isSubmitting: false
+        });
+        dispatch(showSystemNotification({ message: formatMessage(messages.deleteComplete) }));
+        onComplete?.();
+      },
+      (e) => {
+        onSubmittingAndOrPendingChange({
+          isSubmitting: false
+        });
+        const response = e.response?.response ?? e.response;
+        dispatch(
+          showSystemNotification({
+            message: response?.message ?? formatMessage(messages.deleteFailed),
+            options: { variant: 'error' }
+          })
+        );
+      }
+    );
```
This subscribe signature is deprecated (as the editor should have shown you). We must use:
```suggestion
    deleteContentType(site, contentType.id).subscribe({
      next() {
        onSubmittingAndOrPendingChange({
          isSubmitting: false
        });
        dispatch(showSystemNotification({ message: formatMessage(messages.deleteComplete) }));
        onComplete?.();
      },
      error(e) {
        onSubmittingAndOrPendingChange({
          isSubmitting: false
        });
        const response = e.response?.response ?? e.response;
        dispatch(
          showSystemNotification({
            message: response?.message ?? formatMessage(messages.deleteFailed),
            options: { variant: 'error' }
          })
        );
      }
    });
```

rart commented:
Please fix comments on the following PR
